File: monstre/AbiliteMonstre.java
package monstre;

public interface AbiliteMonstre {
    void performAbilite();
}

class Rien implements AbiliteMonstre {
    @Override
    public void performAbilite() {
        System.out.println("Le monstre n'a pas d'abilite!");
    }
}

class PoisonAbility implements AbiliteMonstre {
    @Override
    public void performAbilite() {
        System.out.println("The monster poisons its opponent!");
    }
}

File: Cs/monstre/AbiliteMonstre.cs
using System;

namespace Monstre
{
    public interface IAbiliteMonstre
    {
        void PerformAbilite();
    }

    public class Rien : IAbiliteMonstre
    {
        public void PerformAbilite()
        {
            Console.WriteLine("Le monstre n'a pas d'abilite!");
        }
    }

    public class PoisonAbility : IAbiliteMonstre
    {
        public void PerformAbilite()
        {
            Console.WriteLine("The monster poisons its opponent!");
        }
    }
}
